javaparser-core/src/main/java/com/github/javaparser/ast/Node.java

```diff
@@ -677,4 +678,50 @@ protected SymbolResolver getSymbolResolver() {
     // We need to expose it because we will need to use it to inject the SymbolSolver
     public static final DataKey<SymbolResolver> SYMBOL_RESOLVER_KEY = new DataKey<SymbolResolver>() {
     };
+
+    /**
+     * Walks the AST, calling the consumer for every node.
+     */
+    public void walk(Consumer<Node> consumer) {
+        new TreeVisitor() {
+            @Override
+            public void process(Node node) {
+                consumer.accept(node);
+            }
+        }.visitPreOrder(this);
+    }
+
+    /**
+     * Walks the AST, calling the consumer for every node of type "nodeType".
+     */
+    public <T extends Node> void walk(Class<T> nodeType, Consumer<T> consumer) {
+        new TreeVisitor() {
+            @Override
+            public void process(Node node) {
+                if (nodeType.isInstance(node)) {
+                    consumer.accept(nodeType.cast(node));
+                }
+            }
+        }.visitPreOrder(this);
+    }
+
+    /**
+     * Walks the AST, returning the all nodes of type "nodeType".
+     */
+    public <T extends Node> List<T> find(Class<T> nodeType) {
```

Comment:
Maybe findAll would make it clear that the result is a list

Reply:
Since I'm thinking of adding find(One), you are right.